src/services/builder.cpp

```diff
@@ -380,22 +380,21 @@ void Builder::color_alpha(std::string alpha_)
 {
   auto alpha(alpha_);
   std::string val = this->get_opts()->foreground;
-  if (alpha.find("#") == 0) {
-    if (alpha.size() == 3)
-      this->color(alpha.substr(0, 3) + val.substr(val.size() - 6));
-    else if (alpha.size() == 4)
-      this->color(alpha + alpha.substr(1));
+  if (alpha.find("#") == std::string::npos ) {
+    alpha = "#" + alpha;
+  }
+
+  if (alpha.size() == 4) {
+    // If alpha looks like #123 then we treat as a color and ignore the fg color
+    this->color(alpha + alpha.substr(1));
```

Comment:
This looks like a bug (I know it wasn't added by you).

`#123 != #123123` but `#112233`

`this->color(alpha);` would be correct.